From 99111eba61151fa450f62cefab1a24237c63f184 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date: Tue, 12 Apr 2016 16:07:41 +0200
Subject: [PATCH 2/4] gnutls_packet_get: avoid null pointer dereference on NULL
 input

That is, still allow the function to handle a NULL packet
input but reset the data contents.
---
 lib/gnutls_record.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/gnutls_record.c b/lib/gnutls_record.c
index 422b491..6428459 100644
--- a/lib/gnutls_record.c
+++ b/lib/gnutls_record.c
@@ -1452,6 +1452,7 @@ void gnutls_packet_get(gnutls_packet_t packet, gnutls_datum_t *data, unsigned ch
 		if (data) {
 			data->data = NULL;
 			data->size = 0;
+			return;
 		}
 	}
 
-- 
2.8.0.rc3

